mkt/search/forms.py

```diff
@@ -177,8 +179,17 @@ def clean_device_and_dev(self):
             raise forms.ValidationError('Invalid device or device type.')
 
     def clean_author(self):
-        return self.cleaned_data.get('author', '').lower()
+        author = self.cleaned_data.get('author')
+        if author:
+            return author.lower()
 
     def clean(self):
         self.clean_device_and_dev()
+
+        # Convert empty things to `None`s.
```

Comment:
I could add an explicit check for `0` also, like I do below for False. I agree, this is shady. The other option is to adjust (or add) the `clean_*` methods to return `None` for their specific case.